### PR TITLE
Dwarven Update

### PR DESCRIPTION
**Pull request by r4d6**
## About The Pull Request
This make it so that dwarves get 100 booze-power from The Manly Dorf, even if they have the dwarfism mutation or the Alcohol Tolerance trait. Because it suck that this drink basically become as weak as water.

As an intended side-effect, dwarves now heal brute and burn damage when drinking The Manly Dorf, because it was asked and I liked the idea.

Third and final change for this Dwarven Update : Dwarves speaking in Dwarvish no longer have a bad Scottish accent. So a dwarf can now understand another dwarf.

## Why It's Good For The Game
Better Dwarves, who won't die.

## Changelog
:cl:
Dwarf : Dwarves with Alcohol Tolerance now take full effect from The Manly Dorf
Dwarf : Dwarves now speak normally, without accent when speaking in Dwarvish
Dwarf : Dwarves now heal when drinking The Manly Dorf
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 1]
```diff
@@ -727,9 +727,12 @@ All effects don't start immediately, but rather get worse over time; the rate is
 /datum/reagent/consumable/ethanol/manly_dorf/on_mob_metabolize(mob/living/M)
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		if(H.dna.check_mutation(DWARFISM) || HAS_TRAIT(H, TRAIT_ALCOHOL_TOLERANCE))
+		if(H.dna.check_mutation(DWARFISM) || HAS_TRAIT(H, TRAIT_ALCOHOL_TOLERANCE || is_species(H, /datum/species/dwarf)))
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: Add a `var/real_dorf = isdwarf(H)` line above and use it in place of the two species check here and two lines below, more efficient processing.

File: code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 2]
```diff
@@ -54,21 +54,22 @@ GLOBAL_LIST_INIT(dwarf_last, world.file2list("strings/names/dwarf_last.txt")) //
...
-			if(islist(value)) //Whether they like it or not.
-				value = pick(value) //This could be drastically reduced if needed though.
-
-			message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
-			message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
-			message = replacetextEx(message, " [key]", " [value]") //Also its scottish.
-
-		if(prob(3))
-			message += pick(" By Armok!")
+	if(speech_args[SPEECH_LANGUAGE] != /datum/language/dwarf) // No accent if they speak their language
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species_types/dwarves.dm:57]: Do an early return instead if the types match.

File: code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 3]
```diff
@@ -54,21 +54,22 @@ GLOBAL_LIST_INIT(dwarf_last, world.file2list("strings/names/dwarf_last.txt")) //
...
+		if(message[1] != "*")
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species_types/dwarves.dm:58]: Early return here too.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species_types/dwarves.dm (hunk 2 above)
Comment (r4d6) [reply code/modules/mob/living/carbon/human/species_types/dwarves.dm:57]: Early returns would prevent the rest of the code of running, including parts outside of this `if`

File: code/modules/mob/living/carbon/human/species_types/dwarves.dm (hunk 3 above)
Comment (r4d6) [reply code/modules/mob/living/carbon/human/species_types/dwarves.dm:58]: Same thing here too

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 1 above)
Comment (r4d6) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: Done

File: code/modules/mob/living/carbon/human/species_types/dwarves.dm (hunk 2 above)
Comment (Ghommie) [reply code/modules/mob/living/carbon/human/species_types/dwarves.dm:57]: Oh, so the "By Armok!" bit at the end is intended regardless? Ok, nice.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**